srt-protocol/src/packet/error.rs

```diff
@@ -50,3 +51,37 @@ impl From<io::Error> for PacketParseError {
         PacketParseError::Io(s)
     }
 }
+
+impl Eq for PacketParseError {}
+
+impl PartialEq for PacketParseError {
```

Comment:
Would be good to have a comment here on how this deviates from a derive'd impl

Reply:
I couldn't derive it because io::Error does not implement Eq or PartialEq.

Reply:
I see. I actually feel iffy about making PacketParseError Eq+PartialEq considering io::Error isn't....digging thorugh it seems like the only thing that relies on that is the simulator's `ScheduledInput`, which I think it would be more correct anyways to just manually implement Eq+PartialEq on it to just consider the Instant. 

So my take is we should rip that out and just add a 

```rust
impl PartialEq for ScheduledInput {
    fn eq(&self, other: &Self) -> bool {
        self.0 == other.0
    }
}

impl Eq for ScheduledInput {}
```

Reply:
(imma go ahead and push that and merge this--we can talk if you don't agree)